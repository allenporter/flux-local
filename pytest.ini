[pytest]
log_level = DEBUG
asyncio_mode = auto
