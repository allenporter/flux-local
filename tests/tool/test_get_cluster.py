"""Tests for the flux-local `get cluster` command."""

import pytest

from syrupy.assertion import SnapshotAssertion

from . import run_command


@pytest.mark.parametrize(
    ("args"),
    [
        (["--path", "tests/testdata/cluster"]),
        (["--path", "tests/testdata/cluster2"]),
        (["--path", "tests/testdata/cluster3"]),
        (
            [
                "--path",
                "tests/testdata/cluster3",
                "--sources",
                "cluster=tests/testdata/cluster3",
            ]
        ),
        (["--path", "tests/testdata/cluster4"]),
        (["--path", "tests/testdata/cluster5"]),
        (["--path", "tests/testdata/cluster6"]),
        (["--path", "tests/testdata/cluster7"]),
        (["--all-namespaces", "--path", "tests/testdata/cluster/"]),
        (["--path", "tests/testdata/cluster", "-o", "yaml"]),
        (["--path", "tests/testdata/cluster", "-o", "yaml", "--enable-images"]),
        (["--path", "tests/testdata/cluster8", "-o", "yaml"]),
        (["--path", "tests/testdata/cluster8", "-o", "yaml", "--enable-images"]),
    ],
    ids=[
        "cluster",
        "cluster2",
        "cluster3-no-source",
        "cluster3",
        "cluster4",
        "cluster5",
        "cluster6",
        "cluster7",
        "all-namespaces",
        "yaml-cluster-no-images",
        "yaml-cluster-images",
        "yaml-cluster8-no-images",
        "yaml-cluster8-images"
    ],
)
async def test_get_cluster(args: list[str], snapshot: SnapshotAssertion) -> None:
    """Test test get cluster commands."""
    result = await run_command(["get", "cluster"] + args)
    assert result == snapshot
