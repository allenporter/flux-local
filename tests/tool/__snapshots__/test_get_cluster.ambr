# serializer version: 1
# name: test_get_cluster[all-namespaces]
  '''
  PATH                      KUSTOMIZATIONS    
  tests/testdata/cluster    4                 
  
  '''
# ---
# name: test_get_cluster[cluster2]
  '''
  PATH                       KUSTOMIZATIONS    
  tests/testdata/cluster2    5                 
  
  '''
# ---
# name: test_get_cluster[cluster3-no-source]
  '''
  PATH                       KUSTOMIZATIONS    
  tests/testdata/cluster3    1                 
  
  '''
# ---
# name: test_get_cluster[cluster3]
  '''
  PATH                       KUSTOMIZATIONS    
  tests/testdata/cluster3    2                 
  
  '''
# ---
# name: test_get_cluster[cluster4]
  '''
  PATH                       KUSTOMIZATIONS    
  tests/testdata/cluster4    3                 
  
  '''
# ---
# name: test_get_cluster[cluster5]
  '''
  PATH                       KUSTOMIZATIONS    
  tests/testdata/cluster5    1                 
  
  '''
# ---
# name: test_get_cluster[cluster6]
  '''
  PATH                       KUSTOMIZATIONS    
  tests/testdata/cluster6    2                 
  
  '''
# ---
# name: test_get_cluster[cluster7]
  '''
  PATH                       KUSTOMIZATIONS    
  tests/testdata/cluster7    3                 
  
  '''
# ---
# name: test_get_cluster[cluster]
  '''
  PATH                      KUSTOMIZATIONS    
  tests/testdata/cluster    4                 
  
  '''
# ---
# name: test_get_cluster[yaml-cluster-images]
  '''
  ---
  clusters:
  - path: tests/testdata/cluster
    kustomizations:
    - name: apps
      namespace: flux-system
      path: tests/testdata/cluster/apps/prod
      helm_repos: []
      helm_releases:
      - name: podinfo
        namespace: podinfo
        chart:
          name: podinfo
          repo_name: podinfo
          repo_namespace: flux-system
          repo_kind: HelmRepository
        images:
        - ghcr.io/stefanprodan/podinfo:6.3.2
        - public.ecr.aws/docker/library/redis:7.0.6
      cluster_policies: []
      config_maps:
      - name: podinfo-config
        namespace: podinfo
      secrets: []
    - name: flux-system
      namespace: flux-system
      path: tests/testdata/cluster/clusters/prod
      helm_repos: []
      helm_releases: []
      cluster_policies: []
      config_maps:
      - name: cluster-config
        namespace: flux-system
      secrets: []
    - name: infra-configs
      namespace: flux-system
      path: tests/testdata/cluster/infrastructure/configs
      helm_repos:
      - name: bitnami
        namespace: flux-system
        url: https://charts.bitnami.com/bitnami
        repo_type: default
      - name: podinfo
        namespace: flux-system
        url: oci://ghcr.io/stefanprodan/charts
        repo_type: oci
      - name: weave-charts
        namespace: flux-system
        url: oci://ghcr.io/weaveworks/charts
        repo_type: oci
      helm_releases: []
      cluster_policies:
      - name: test-allow-policy
      config_maps: []
      secrets: []
    - name: infra-controllers
      namespace: flux-system
      path: tests/testdata/cluster/infrastructure/controllers
      helm_repos: []
      helm_releases:
      - name: kyverno
        namespace: kyverno
        chart_ref:
          kind: OCIRepository
          name: kyverno
          namespace: flux-system
      - name: metallb
        namespace: metallb
        chart:
          name: metallb
          repo_name: bitnami
          repo_namespace: flux-system
          repo_kind: HelmRepository
        images:
        - docker.io/bitnami/metallb-controller:0.13.7-debian-11-r29
        - docker.io/bitnami/metallb-speaker:0.13.7-debian-11-r28
      - name: weave-gitops
        namespace: flux-system
        chart:
          name: weave-gitops
          repo_name: weave-charts
          repo_namespace: flux-system
          repo_kind: HelmRepository
        images:
        - ghcr.io/weaveworks/wego-app:v0.24.0
      cluster_policies: []
      config_maps: []
      secrets: []
  
  '''
# ---
# name: test_get_cluster[yaml-cluster-no-images]
  '''
  ---
  clusters:
  - path: tests/testdata/cluster
    kustomizations:
    - name: apps
      namespace: flux-system
      path: tests/testdata/cluster/apps/prod
      helm_repos: []
      helm_releases:
      - name: podinfo
        namespace: podinfo
        chart:
          name: podinfo
          repo_name: podinfo
          repo_namespace: flux-system
          repo_kind: HelmRepository
      cluster_policies: []
      config_maps:
      - name: podinfo-config
        namespace: podinfo
      secrets: []
    - name: flux-system
      namespace: flux-system
      path: tests/testdata/cluster/clusters/prod
      helm_repos: []
      helm_releases: []
      cluster_policies: []
      config_maps:
      - name: cluster-config
        namespace: flux-system
      secrets: []
    - name: infra-configs
      namespace: flux-system
      path: tests/testdata/cluster/infrastructure/configs
      helm_repos:
      - name: bitnami
        namespace: flux-system
        url: https://charts.bitnami.com/bitnami
        repo_type: default
      - name: podinfo
        namespace: flux-system
        url: oci://ghcr.io/stefanprodan/charts
        repo_type: oci
      - name: weave-charts
        namespace: flux-system
        url: oci://ghcr.io/weaveworks/charts
        repo_type: oci
      helm_releases: []
      cluster_policies:
      - name: test-allow-policy
      config_maps: []
      secrets: []
    - name: infra-controllers
      namespace: flux-system
      path: tests/testdata/cluster/infrastructure/controllers
      helm_repos: []
      helm_releases:
      - name: kyverno
        namespace: kyverno
        chart_ref:
          kind: OCIRepository
          name: kyverno
          namespace: flux-system
      - name: metallb
        namespace: metallb
        chart:
          name: metallb
          repo_name: bitnami
          repo_namespace: flux-system
          repo_kind: HelmRepository
      - name: weave-gitops
        namespace: flux-system
        chart:
          name: weave-gitops
          repo_name: weave-charts
          repo_namespace: flux-system
          repo_kind: HelmRepository
      cluster_policies: []
      config_maps: []
      secrets: []
  
  '''
# ---
# name: test_get_cluster[yaml-cluster8-images-allow-secrets]
  '''
  ---
  clusters:
  - path: tests/testdata/cluster8
    kustomizations:
    - name: apps
      namespace: flux-system
      path: tests/testdata/cluster8/apps
      helm_repos:
      - name: podinfo
        namespace: flux-system
        url: oci://ghcr.io/stefanprodan/charts
        repo_type: oci
      - name: tailscale
        namespace: flux-system
        url: https://pkgs.tailscale.com/helmcharts
        repo_type: default
      helm_releases:
      - name: podinfo
        namespace: podinfo
        chart:
          name: podinfo
          repo_name: podinfo
          repo_namespace: flux-system
          repo_kind: HelmRepository
        images:
        - ghcr.io/stefanprodan/podinfo:6.5.4
        - public.ecr.aws/docker/library/redis:7.0.6
      - name: tailscale-operator
        namespace: network
        chart:
          name: tailscale-operator
          repo_name: tailscale
          repo_namespace: flux-system
          repo_kind: HelmRepository
        images:
        - tailscale/k8s-operator:v1.68.1
      cluster_policies: []
      config_maps:
      - name: podinfo-values
        namespace: podinfo
      secrets:
      - name: podinfo-tls-values
      images:
      - alpine
      - busybox
    - name: flux-system
      namespace: flux-system
      path: tests/testdata/cluster8/cluster
      helm_repos: []
      helm_releases: []
      cluster_policies: []
      config_maps: []
      secrets: []
  
  '''
# ---
# name: test_get_cluster[yaml-cluster8-images]
  '''
  ---
  clusters:
  - path: tests/testdata/cluster8
    kustomizations:
    - name: apps
      namespace: flux-system
      path: tests/testdata/cluster8/apps
      helm_repos:
      - name: podinfo
        namespace: flux-system
        url: oci://ghcr.io/stefanprodan/charts
        repo_type: oci
      - name: tailscale
        namespace: flux-system
        url: https://pkgs.tailscale.com/helmcharts
        repo_type: default
      helm_releases:
      - name: podinfo
        namespace: podinfo
        chart:
          name: podinfo
          repo_name: podinfo
          repo_namespace: flux-system
          repo_kind: HelmRepository
        images:
        - ghcr.io/stefanprodan/podinfo:6.5.4
        - public.ecr.aws/docker/library/redis:7.0.6
      - name: tailscale-operator
        namespace: network
        chart:
          name: tailscale-operator
          repo_name: tailscale
          repo_namespace: flux-system
          repo_kind: HelmRepository
        images:
        - tailscale/k8s-operator:v1.68.1
      cluster_policies: []
      config_maps:
      - name: podinfo-values
        namespace: podinfo
      secrets: []
      images:
      - alpine
      - busybox
    - name: flux-system
      namespace: flux-system
      path: tests/testdata/cluster8/cluster
      helm_repos: []
      helm_releases: []
      cluster_policies: []
      config_maps: []
      secrets: []
  
  '''
# ---
# name: test_get_cluster[yaml-cluster8-no-images]
  '''
  ---
  clusters:
  - path: tests/testdata/cluster8
    kustomizations:
    - name: apps
      namespace: flux-system
      path: tests/testdata/cluster8/apps
      helm_repos:
      - name: podinfo
        namespace: flux-system
        url: oci://ghcr.io/stefanprodan/charts
        repo_type: oci
      - name: tailscale
        namespace: flux-system
        url: https://pkgs.tailscale.com/helmcharts
        repo_type: default
      helm_releases:
      - name: podinfo
        namespace: podinfo
        chart:
          name: podinfo
          repo_name: podinfo
          repo_namespace: flux-system
          repo_kind: HelmRepository
      - name: tailscale-operator
        namespace: network
        chart:
          name: tailscale-operator
          repo_name: tailscale
          repo_namespace: flux-system
          repo_kind: HelmRepository
      cluster_policies: []
      config_maps:
      - name: podinfo-values
        namespace: podinfo
      secrets: []
    - name: flux-system
      namespace: flux-system
      path: tests/testdata/cluster8/cluster
      helm_repos: []
      helm_releases: []
      cluster_policies: []
      config_maps: []
      secrets: []
  
  '''
# ---
